### PR TITLE
Ohad k8s pool changes

### PR DESCRIPTION
**Pull request by nb-ohad**
### Explain the changes
1.  Fix unmanaged pool deployment issues and UI
2. Fix - deployment script not deleting PVCs managed by the system
3. Managed hosts pools improvements

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

File: src/server/system_services/pool_server.js [hunk 1]
```diff
@@ -460,12 +497,12 @@ async function scale_hosts_pool(req) {
         update: {
             pools: [{
                 _id: pool._id,
+                'hosts_pool_info.initialized': true,
```
Comment (dannyzaken) [inline src/server/system_services/pool_server.js]: `initialized` should be set after scale succeeds (inside `scale_hosts_pool_job`) and not here

File: src/server/system_services/pool_server.js [hunk 2]
```diff
@@ -95,12 +97,24 @@ async function _init() {
...
+                    }
+                }
 
             } else if (configured_host_count < actual_host_count) {
                 const delete_count = actual_host_count - configured_host_count;
                 await nodes_client.instance().delete_hosts_by_pool(pool.name, system._id, delete_count);
-                await wait_for_host_count_to_stabilize(system, pool, configured_host_count);
+                await wait_for_host_count_to_stabilize(system._id, pool._id, configured_host_count);
                 await pool_ctrl.scale(configured_host_count);
             }
```
Comment (dannyzaken) [inline src/server/system_services/pool_server.js]: need to check here if not initialize and set `initialized=true`

File: src/server/system_services/pool_server.js [hunk 3]
```diff
@@ -475,36 +512,79 @@ async function scale_hosts_pool(req) {
...
+    } else if (host_count_diff < 0) {
+        // Scale down (delete hosts)
+        dbg.log0(`scale_hosts_pool: deleting ${-host_count_diff} hosts`);
+        await nodes_client.instance().delete_hosts_by_pool(pool.name, system._id, -host_count_diff);
+        await wait_for_host_count_to_stabilize(system._id, pool._id, configured_host_count);
+
+        // No need to wait for hosts because the  the scale just cleanup the underlaying
+        // agents after the hosts deletion.
+        await pool_ctrl.scale(configured_host_count);
     }
```
Comment (dannyzaken) [inline src/server/system_services/pool_server.js]: set initialized here if needed

File: src/server/system_services/pool_server.js [hunk 4]
```diff
@@ -549,6 +629,7 @@ async function delete_hosts_pool(req, pool) {
             update: {
                 pools: [{
                     _id: pool._id,
+                    'hosts_pool_info.initialized': true,
```
Comment (dannyzaken) [inline src/server/system_services/pool_server.js]: no need to set initialized on delete

**Review by dannyzaken — CHANGES_REQUESTED** [review]

File: src/deploy/NVA_build/noobaa_pool.yaml [hunk 5]
```diff
@@ -41,12 +40,12 @@ spec:
     spec:
       containers:
         - name: noobaa-agent
-          readinessProbe:
-            # https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes
-            # ready when s3 port is open
-            tcpSocket:
-              port: 6001
-            timeoutSeconds: 5
+          # readinessProbe:
```
Comment (dannyzaken) [inline src/deploy/NVA_build/noobaa_pool.yaml]: why did you eventually remove it? thought it was working with the probe

**Review by nimrod-becker — APPROVED** [review]

**Pull request merged**